app/main.py

```diff
@@ -1,34 +1,109 @@
 class Deck:
-    def __init__(self, row, column, is_alive=True):
-        pass
+    def __init__(self, row: int, column: int, is_alive: bool = True) -> None:
+        self.row = row
+        self.column = column
+        self.is_alive = is_alive
 
 
 class Ship:
-    def __init__(self, start, end, is_drowned=False):
-        # Create decks and save them to a list `self.decks`
-        pass
+    def __init__(
+        self,
+        start: tuple,
+        end: tuple,
+        is_drowned: bool = False
+    ) -> None:
+        self.is_drowned = is_drowned
+        self.decks = []
+        if start == end:
+            self.decks.append(Deck(start[0], start[1]))
+        elif start[0] == end[0] and start[1] != end[1]:
+            for num in range(end[1] - start[1] + 1):
+                self.decks.append(Deck(start[0], start[1] + num))
+        else:
+            for num in range(end[0] - start[0] + 1):
+                self.decks.append(Deck(start[0] + num, start[1]))
 
-    def get_deck(self, row, column):
-        # Find the corresponding deck in the list
-        pass
+    def get_deck(self, row: int, column: int) -> Deck:
+        for deck in self.decks:
+            if deck.row == row and deck.column == column:
+                return deck
 
-    def fire(self, row, column):
-        # Change the `is_alive` status of the deck
-        # And update the `is_drowned` value if it's needed
-        pass
+    def fire(self, row: int, column: int) -> bool:
+        deck = self.get_deck(row, column)
+        if not self.is_drowned and deck.is_alive:
```

Comment:
let's swap conditions because it's easier to read the condition without NOT first